### PR TITLE
Events

### PR DESCRIPTION
**Pull request by Calebc4554**
# Description  
- What does this PR do?  
  Builds the full Events page UI using real Supabase data. Adds a working "Register" button that saves each user's registration in the `event_registrations` table, updates their total points in the `users` table, and keeps the "Registered" state even after reloading the page.  

- Why is it necessary or valuable?  
 Connects the frontend to the backend so users can sign up for events and earn points. This sets up the main system for tracking event participation and giving credit for it.

- What is intentionally left out and will be done in a later PR?  
  - Add to Microsoft Outlook Calendar using Microsoft Graph API  
  - Admin/host tools for creating and editing events  

---

# Milestones / Related Work  
- Feature or user story this contributes to:  
  - [Earn Points by Registering for Events](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=id.6r25kxcyl8y8)  
  - [View MICS Events](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=id.m5ze28nmtezt)  

- Milestone or version target:  
  - [2.3 Users can view all of the events](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=kix.7daru3xyvrxg)  
  - [2.4 Users can register for events and get points for it](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=id.4fdesxkbau2t)  

---

# Resources and References  
- Tutorials, documentation, or code examples used:  
  - https://supabase.com/docs  

---

# Test Plan  
- How was this tested?  
  - Manually tested by registering and unregistering from different events  
  - Checked Supabase to make sure user points updated correctly  
  - Reloaded page to confirm button state stays updated  

- Screenshots or recordings:  
[Loom Demo Link](https://www.loom.com/share/5796ad0bee1543f891edf46f1294ed28?sid=d0b61809-9de4-4462-a469-121658b1e43a)

- Seed or mock data used:  
  - Test users and events stored in Supabase  
  - Manually added entries to `event_registrations` for testing  

## Edge Cases Tested  
- Clicking register/unregister quickly  
- Reloading the page after registering  

---

# Additional Notes  
- Known limitations:  
  - "Add to Calendar" button doesn’t work yet  

- Planned follow-up PRs:  
  - Outlook calendar integration  
  - Better UI styling and responsiveness

### CODE REVIEW

File: frontend/src/components/Events.jsx [hunk 1]
```diff
@@ -1,20 +1,101 @@
...
+					<p className="text-center text-gray-600">Loading events…</p>
+				) : (
+					<div className="space-y-6">
+						{events.map((e) => {
+							const isReg = registered.has(e.id);
+							return (
+								<div key={e.id} className="bg-white rounded-lg shadow-md p-6 flex flex-col">
+									<h2 className="text-xl font-semibold">{e.title}</h2>
+									<div className="text-gray-500 text-sm mt-1">
+										{new Date(e.date).toLocaleString(undefined, {
```
Comment (weibinma123) [inline frontend/src/components/Events.jsx:72]: Wondering if we can also add the time zone here? it would be easier for users to know the exact time.

File: frontend/src/components/Events.jsx [hunk 2]
```diff
@@ -1,20 +1,101 @@
...
+										})}
+									</div>
+									<div className="text-gray-500 text-sm">{e.location}</div>
+									<p className="mt-4 text-gray-700">{e.description}</p>
+									<div className="mt-4 text-sm text-gray-600">
+										Created by <span className="font-medium text-gray-800">{e.users?.name ?? "Unknown"}</span>
+									</div>
+									<div className="mt-6 flex gap-4">
+										<button onClick={() => toggleRegister(e.id)} className={`text-sm font-medium py-2 px-4 rounded transition ${isReg ? "bg-green-500 hover:bg-green-600" : "bg-blue-500 hover:bg-blue-600"} text-white`}>
+											{isReg ? "Registered" : "Register"}
```
Comment (weibinma123) [inline frontend/src/components/Events.jsx:88]: Once "Registered", why do we still allow users to click to switch back to "Register"? Users probably could be confused as they're unsure if registered successfully or not.

File: frontend/src/components/Events.jsx (hunk 1 above)
Comment (Calebc4554) [reply frontend/src/components/Events.jsx:72]: Yes, definitely! Just implemented!
